homeassistant/components/channels/media_player.py

```diff
@@ -67,6 +79,10 @@
     {vol.Required(ATTR_SECONDS): vol.Coerce(int)}
 )
 
+CHANNELS_NAVIGATE_SCHEMA = CHANNELS_SCHEMA.extend(
+    {vol.Required(ATTR_NAME): vol.Coerce(str)}
+)
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
For your config entry, rename this to 
```suggestion
async def async_setup_platform(hass, entry, async_add_entities):
```

And update references to `async_add_entities` accordingly. 

All calls to `hass.services.register` should become `hass.services.async_register` to make it work again. 

The `platform.async_register_entity_service` takes care of all the things you do with keeping track of devices for the service to target and calling the right function.